plugins/modules/fusion_array.py

```diff
@@ -219,7 +238,7 @@ def delete_array(module, fusion):
             module.fail_json(
                 msg="Array {0} creation failed.: {1}".format(module.params["name"], err)
             )
-    module.exit_json(changed=changed)
+    return True
```

Comment:
Please use the core ansible exit_json when leaving a function. This will downstream affects for Ansible if you don't.

Reply:
I understand that `module.exit_json()` returns results from the module to upstream Ansible, the call was just moved down the stream into `main()` and is still called. The reasoning behind this change is that quite a create functions for some resources in the SDK can only set a few properties and the rest are left at default. The rest of them have to be changed via patch routines, which is basically what `update_array()` and other `update_*()` functions do. So I've separated the concerns of `create_array()` and `update_array()` and `main()` now calls them in succession and calls `module.exit_json()` with their merged result. A lot of the patch calls would be have to otherwise duplicated.

Reply:
Same design is used in [`fusion_volume`](https://github.com/Pure-Storage-Ansible/Fusion-Collection/blob/master/plugins/modules/fusion_volume.py#L261) ([here](https://github.com/Pure-Storage-Ansible/Fusion-Collection/blob/master/plugins/modules/fusion_volume.py#L562) is how `create_volume()` and `update_volume()` are called)

Reply:
Can you test to make sure that these functions are still idempotent. Run your test playbooks twice to ensure they return the correct responses on the second run, ie no change

Reply:
I got an idea how to test the modules better and found a few bugs which I've fixed. Otherwise the array and volume modules are both idempotent in the sense they properly identify when they need to actually change anything and do it only then and report 'ok' otherwise. The create + update flow works correctly.